### ISSUE TITLE
datasets

### ISSUE THREAD

**Issue by 1044197988** [DequanZhu/FaceNet-and-FaceLoss-collections-tensorflow2.0#3]
用LFW数据集，代码label错误。
这个label是需要数字编码吗？  如 'dgasdg' = 1

感谢你的代码                     

最近，我使用你最新的代码，lfw进行了label编码,总共5749个类
这里进行了修改
但是训练的时候loss从21降到20左右突然降到8，然后就不收敛了。
acc一直是0，我检查了各个变量的形状，感觉很正确
还有tfrecord的image与label也都正确
image是经过人脸对齐的
模型用的backone为xception
是不是训练的时间很短的原因，要一直训练很多周期才有效果呢？

**Comment by DequanZhu** [comment]
对的，label是用数字编码的，我的代码中是根据每一类的文件夹名提取相应的label的。你的loss是用的什么的呢，我在loss.py中写了好几个loss，train.py中的loss默认是ArcFaceSoftmaxLinear，这个你要自己修改成想要的loss，还有一些学习率之类的。另外我没有训练过模型，慎重使用，只是提供代码参考。

**Comment by 1044197988** [comment]
loss用的ArcFaceSoftmaxLinear ，优化器adam
我参考 https://github.com/4uiiurz1/keras-arcface
使用相同的参数 与 embedding、label
得到的值如下

tf.Tensor(
[[-1.85826263e+02  1.78392362e-02  3.18274021e-01 -2.43931785e-01
   4.43012148e-01]
 [ 4.65939552e-01 -1.87115738e+02 -1.68064594e-01  3.11165512e-01
  -2.62193173e-01]
 [ 6.80084586e-01 -1.71573654e+02 -1.87849730e-01 -8.77188817e-02
   1.97854653e-01]
 [-3.73935848e-01  3.32845032e-01 -4.68550563e-01 -2.07526291e+02
  -3.76850724e-01]
 [ 2.59314895e-01 -4.94366974e-01  1.12484634e-01 -3.16056579e-01
  -1.51565826e+02]
 [-3.28895420e-01 -1.87781751e-01 -2.36562775e+02  4.37414944e-02
  -2.61997432e-01]
 [-1.71239868e+02 -1.49381429e-01  1.43269405e-01 -2.32902199e-01
   5.99399269e-01]
 [-2.58152932e-01 -6.74506068e-01 -4.64719057e-01 -3.14436138e-01
  -1.72879257e+02]], shape=(8, 5), dtype=float32)

keras-arcface：
tf.Tensor(
[[1.4199907e-31 1.4198779e-22 2.1054075e-18 7.8027530e-17 1.0000000e+00]
 [9.9985850e-01 4.9019848e-17 1.4142200e-04 3.5029832e-17 1.8676406e-14]
 [1.1358455e-21 7.0185430e-10 5.7009778e-17 1.7849522e-11 1.0000000e+00]
 [9.9999988e-01 9.4455051e-18 1.3115114e-07 6.1613063e-26 2.1979572e-16]
 [3.5800247e-06 9.9999642e-01 1.5133526e-12 2.4137991e-17 1.8823791e-18]
 [5.0200590e-05 9.1372615e-01 4.2128143e-10 8.6223662e-02 4.5481355e-16]
 [2.1673259e-31 1.7637790e-12 4.7586875e-13 1.0179751e-21 1.0000000e+00]
 [1.2029181e-09 9.9629670e-01 1.5113739e-05 3.1467299e-15 3.6882248e-03]], shape=(8, 5), dtype=float32)
他的arcface最后使用了softmax
并且是独热编码的Label
虽然初始化不同，但是可以看到他的无负值
我认为可能是Loss的计算出了问题？

**Comment by DequanZhu** [comment]
试一试其他loss呢？

**Comment by 1044197988** [comment]
其他的也不行，我正尝试重写ArcFace

**Comment by 1044197988** [comment]
如果公式没错的话
只需要在SparseCategoricalCrossentropy(from_logits=True)加上from_logits=True就可以了
感谢代码
默认为False

**Comment by LawsonLv** [comment]
from_logits=True; acc还是一直是0
